syntax = "proto3";

service HttpProxy {
    rpc RunStory (Request) returns (stream ResponseCommand) {
    }
}

message ResponseCommand {
    string command = 1;
    map<string, string> args = 2;
}

message Request {
    string path = 1;
    string story_name = 2;
    repeated Header headers = 3;
    bytes body = 4;
    string method = 5; // POST/GET/etc
    string hostname = 6;
    string app_id = 7;
    string json_context = 8;
    string json_environment = 9;
    string block = 10;
    string start = 11;
}

message Header {
    string key = 1;
    string value = 2;
}